### ISSUE TITLE
Hash#to_h with a block

### ISSUE THREAD

**Issue by zverok** [marcandre/backports#132]
Native in Ruby 2.6:
```ruby
2.6.0-rc2 :001 > {a: 1, b: 2, c: 3}.to_h {|k,v| [k, v * 2]}
# => {:a=>2, :b=>4, :c=>6} 
```
With backports on 2.4:
```ruby
2.4.2 :002 > {a: 1, b: 2, c: 3}.to_h {|k,v| [k, v * 2]}
# => {:a=>1, :b=>2, :c=>3} 
```

**Comment by marcandre** [comment]
Oh, thanks for the report. I indeed forgot to create `backports/2.6.0/hash/to_h`.

I'm travelling right now, should have the time within a week.

**Comment by marcandre** [comment]
Released as v3.14. Thanks for the report.
